src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -177,42 +177,6 @@ public String toString() {
             "]";
   }
 
-  private static <T> String dotShortName(T t) {
-    String s = t.toString();
-    int n1 = s.lastIndexOf('.') + 1;
-    int n2 = s.indexOf('(');
-    return s.substring(n1, n2);
-  }
-
-  /**
-   * @return a .dot file (GraphViz) version of this graph.
-   */
-  public String toDot() {
```

Comment:
Could you try to export the GraphViz feature somewhere outside the class (maybe by extension)? It is helpful for the debug

Reply:
Fixed this. This unexposed functionality is now exposed via https://github.com/cbeust/testng/pull/1829

Reply:
👍